RDKBACCL-1329 : TLV Length correction for AP Autoconfiguration M2

SachinMChougale opened the pull request:
1. Connection type flag len changed to 1B
2. Primary Device type len changed to 8B
3. OS Version Len changed to 4B

amarnathhullur commented on src/em/config/em_configuration.cpp:
```diff
@@ -2148,7 +2150,8 @@ unsigned short em_configuration_t::create_m2_msg(unsigned char *buff, em_haul_ty
+    // Os version should be 4 bytes
+    size = 4;
     attr->len = htons(size);
     memcpy(attr->val, &rf_band, sizeof(attr->val));
```
The sizeof(attr->val)) would return 0 since the definition of attr->val is unsigned char val[0], so we don't seem to be copying anything in attr->val but just incrementing by 4 bytes.
This seems like wrong code earlier and needs to be changed where ever sizeof(attr->val) is being used for memcpy as it would not copy anything to attr->val

SachinMChougale committed:
RDKBACCL-1329 : TLV Length cnd Value correction
1. Connection type flag len changed to 1B
2. Primary Device type len changed to 8B
3. OS Version Len changed to 4B

Signed-off-by: cm.sachin <cm.sachin@capgemini.com>
